src/node/internal/internal_http_server.ts

```diff
@@ -211,7 +209,7 @@ export class Server
     this.port = this.#findSuitablePort(Number(options.port));
     portMapper.set(this.port, { fetch: this.#onRequest.bind(this) });
     queueMicrotask(() => {
-      callback?.();
+      this.emit('listening');
```

Comment:
Just noting here, we will want to make sure documentation for this captures the fact that if any of the `listening` handlers (here and in any of the other `queueMicrotask(...)` instances here... if the `listening` handlers throw an error, that will end up being reported to `reportError(...)` and will cause the globalThis `error` event to be triggered.